packages/block-editor/src/components/iframe/index.js

```diff
@@ -126,21 +128,29 @@ function setBodyClassName( doc ) {
 	}
 }
 
-/**
- * Sets the document head and default styles.
- *
- * @param {Document} doc  Document to set the head for.
- * @param {string}   head HTML to set as the head.
- */
-function setHead( doc, head ) {
-	doc.head.innerHTML =
-		// Body margin must be overridable by themes.
-		'<style>body{margin:0}</style>' + head;
+function useParsedAssets( html ) {
+	return useMemo( () => {
+		const doc = document.implementation.createHTMLDocument( '' );
+		doc.body.innerHTML = html;
+		return Array.from( doc.body.children );
+	}, [ html ] );
 }
 
-function Iframe( { contentRef, children, head, headHTML, ...props }, ref ) {
-	const [ iframeDocument, setIframeDocument ] = useState();
+async function loadScript( doc, { id, src } ) {
+	return new Promise( ( resolve, reject ) => {
+		const script = doc.createElement( 'script' );
+		script.id = id;
+		script.src = src;
+		script.onload = () => resolve();
```

Comment:
If a script does not have an `src` (eg. in my local dev install it adds `<script id="wp-polyfill-js-after"></script>`), then this never resolves so any scripts following are not loaded.

Changing this to 
```js
if ( src ) {
    script.src = src;
    script.onload = () => resolve();
    script.onerror = () => reject();
} else {
    resolve();
}
```
fixed this for me and everything ran as expected.

Reply:
Ok, I'll change it. Interesting that inline scripts are parsed and loaded synchronously.